### PR TITLE
Various test flakiness fixes

### PR DESCRIPTION
**Pull request by Quexington**
This adds some fixes to make tests theoretically less flaky (important for upstream PRs)

### CODE REVIEW

**Review by AmineKhaldi — DISMISSED** [review]

**Review by AmineKhaldi — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Comment by Quexington** [comment]
@arvidn had comments for this on a different PR this should wait for his approval as well

**Pull request merged**
